src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -59,13 +59,24 @@ public async Task<Solution> CreateSolutionAsync(Checksum newSolutionChecksum, Ca
                     // if needed again later.
                     solution = solution.WithoutFrozenSourceGeneratedDocuments();
 
-                    var oldSolutionCompilationChecksums = await solution.CompilationState.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
                     var newSolutionCompilationChecksums = await _assetProvider.GetAssetAsync<SolutionCompilationStateChecksums>(
                         assetHint: AssetHint.None, newSolutionChecksum, cancellationToken).ConfigureAwait(false);
-                    var oldSolutionChecksums = await solution.CompilationState.SolutionState.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
+
                     var newSolutionChecksums = await _assetProvider.GetAssetAsync<SolutionStateChecksums>(
                         assetHint: AssetHint.None, newSolutionCompilationChecksums.SolutionState, cancellationToken).ConfigureAwait(false);
 
+                    // If we are only syncing over a project cone, then only actually compare the respective cones on
+                    // the host and remote side here.
+                    var projectConeId = newSolutionChecksums.ProjectConeId;
```

Comment:
the remote site is the most interesting.  we now see when we're syncing a project cone and update our logic accordingly.